### ISSUE TITLE
[Deque Analysis] Search Results "critical" accessibility issues

### ISSUE THREAD

**Issue by tdonohue** [DSpace/dspace-angular#1168]
> Critical: This issue results in blocked content for people with disabilities, and will definitely prevent them from accessing fundamental features or content. This type of issue puts your organization at risk. Prioritize fixing as soon as possible, within the week if possible. Remediation should be a top priority. 

## Deque Analysis Summary

A total of 24 Critical issues exist in the Search Results (https://demo7.dspace.org/search?spc.page=1&query=test) . They are summarized here:
1. (20 issues) All Filter checkboxes are missing labels.  The values next to the checkboxes are improperly in `<span>` tags instead of `<label>` tags which reference the checkbox input field.  See for example: https://www.a11ymatters.com/pattern/checkbox/#associating-the-label-with-the-input
    * Alternatively, they can be kept as `<span>`, but they need an `id` attribute which can be referenced by `aria-labelledby` (e.g. `<input type="checkbox" aria-labelledby="text"><span id="text">Text</span>`
    * Issue IDs: 468524, 468525, 468526, 468527, 468528, 468529, 468530, 468531, 468532, 468533, 468534, 468535, 468536, 468537, 468538, 468539, 468540, 468541, 468542, 468543
2. (ID 468956) All Filter free-text entry fields are missing labels (e.g. "Author name", "Subject", Date fields, etc).  We should either add a `label` above each, or add a `aria-label` attribute to describe them.
     * In a separate "serious" ticket (ID 468933), Deque recommends that these fields have "visible labels" as their purpose is not clear.  Perhaps we should create a small label saying "Search filter values" or similar?
3. (ID 468903) Expandable/Accordion sections (filters) are not keyboard accessible. There's no way to expand them using just keyboard.  Here's an example of how this might be achieved: https://www.w3.org/TR/wai-aria-practices-1.1/examples/accordion/accordion.html  (Others may exist as well, this was just the first I found that seemed similar)
4. (ID 468906) "Show More" and "Collapse" links (in filter sidebar) have no `href` value.
    * Fix this issue by providing an HREF value (even if JavaScript event handlers are used) for the link. (Note: Ensure that if JavaScript event handlers are used, they allow the link to be activated with the Enter key in addition to mouse click.)
    * e.g. `<a href=quickview.html">Quick view</a>`
    * e.g. `<a href="javascript:void(0);">Quick view</a>`
5. (ID 469004) Date slider increase/decrease buttons are missing a focus indicator. All focusable elements MUST have a visual focus indicator when in focus.
    * Common keyboard focus indicator solutions include (choose one): (1) A 2px box around the focused element, (2) A change in the background color of the focused element (3) The addition of an icon, such as an arrow, next to a menu item (4) The addition of a thick line under or next to the focused element (5) A distinctive change to the text of the component such as making it bold and/or underlined (6) Use the default browser focus indicator. NOTE: Since each browser has its own default focus indicator style, check each of the major browsers (Chrome, Firefox, Edge, IE, Safari) to ensure the default focus indicator is visible as you tab through the page. 
     * See also https://dequeuniversity.com/class/input-methods2/keyboard-input/visual-focus-indicator

The full list of these issues (which includes links for more info) can be found at this link (requires login): https://axeauditor.dequecloud.com/test-run/0856438a-a19a-11eb-bc31-b7d5be387c86/issues?activeTab=dt-issue&page=0&pageSize=100&sortField=ordinal&sortDir=asc&row=9&filter%5Bseverity%5D=4&filter%5Bpage_number%5D=1

## More Information / Tools

* Consider using [Deque's free Chrome plugin](https://chrome.google.com/webstore/detail/axe-devtools-web-accessib/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US) to check your work

**Comment by artlowel** [comment]
>1. (20 issues) All Filter checkboxes are missing labels.

- Adding a `<label for="${id}">` is difficult because we have no good way to have id's that are sure to be unique page-wide in angular. Instead we'll add `<label>` tags around both the checkbox and the span, which is also valid HTML. I've tested it and it's approved by Deque's chrome extension as well.

>2. (ID 468956) All Filter free-text entry fields are missing labels

- We'll add a "Search filter values" label above each field

>3. (ID 468903) Expandable/Accordion sections (filters) are not keyboard accessible.

- We'll turn the toggle icons in to buttons so they can be selected with the keyboard and get an outline
- add aria-control and aria-expanded attributes to the toggle button
- give the entire filter section an outline if it has the focus
- add an id to the section based on a combination of the pagination id and the filter name so aria-control can refer to it

>4. (ID 468906) "Show More" and "Collapse" links (in filter sidebar) have no `href` value.

- We'll add a href attribute to those links

>5. (ID 469004) Date slider increase/decrease buttons are missing a focus indicator.

- We'll add an outline to the handle if it has the focus. 
- Also, while that slider is keyboard accessible out of the box, it is very hard to use because it will update the page immediately after every change. That's very slow. We'll add a debounce to the slider so that it waits until there's no more change for e.g. 500ms before sending an update to the server.

This will take an estimated **8 hours**
